### PR TITLE
 Add termination reason `SUBSCRIPTION_DELETED` when subscription deleted by the user

### LINKED ISSUE #117: Send a CloudEvent when DELETE `/subscriptions/{subscriptionId}`
**Problem description**
Currently there is no description about a CloudEvent when the subscription will be deleted by a User via  DELETE `/subscriptions/{subscriptionId}`.

**Possible evolution**

- Update the description of  `DELETE /subscriptions/{subscriptionId´ to mention that a CloudEvent will be send as well.
- Add a `terminationReason` like `DELETE_REQUESTED`

see how QoD v0.10.0 is doing it when the user requested a deletion before subscription-expired.

https://github.com/camaraproject/QualityOnDemand/blob/v0.10.0/code/API_definitions/qod-api.yaml#L918

### PR DESCRIPTION
**Pull request by maxl2287**
#### What type of PR is this?

Add one of the following kinds:

* enhancement/feature

#### What this PR does / why we need it:
Adds a new termination-reason for "subscription-ends" - CloudEvent, when the user deletes the subscriptions via 
`DELETE /subscriptions/{subscriptionId}`

#### Which issue(s) this PR fixes:

Fixes #117

### CODE REVIEW

**Comment by akoshunyadi** [comment]
We should wait until the problem is solved in commonalities, we don't want to have proprietary solutions in each APIs

**Comment by akoshunyadi** [comment]
The discussion about subscription lifecycle management in commonalities: https://github.com/camaraproject/Commonalities/issues/153

**Comment by maxl2287** [comment]
waiting for megalinter fixes which are already done by #146

**Comment by maxl2287** [comment]
@camaraproject/device-status_maintainers 
This can be reviewed and possibly merged after https://github.com/camaraproject/Commonalities/pull/238 is merged

File: code/API_definitions/device-reachability-status-subscriptions.yaml [hunk 1]
```diff
@@ -36,7 +36,11 @@ info:
 
     Note: Additionally to these list, ``org.camaraproject.reachability-status-subscriptions.v0.subscription-ends`` notification `type` is sent when the subscription ends.
     This notification does not require dedicated subscription.
-    It is used when the subscription expire time (optionally set by the requester) has been reached or if the API server has to stop sending notification prematurely.
+
+    It is used when:
+      - the subscription expire time (optionally by the requester) has been reached
```
Comment (akoshunyadi) [inline code/API_definitions/device-reachability-status-subscriptions.yaml:41]: Maybe better: "It is used in following cases:" to make it clear that the cases are not AND linked

**Review by akoshunyadi — APPROVED** [review]

**Comment by maxl2287** [comment]
The related Communality issue is closed.
So we are ready to merge this one here.

**Review by bigludo7 — APPROVED** [review]
LGTM/

**Review by sachinvodafone — DISMISSED** [review]

**Review by bigludo7 — APPROVED** [review]
LGTM/

**Review by akoshunyadi — APPROVED** [review]

**Review by sachinvodafone — APPROVED** [review]

**Pull request merged**